packages/datadog_flutter_plugin/lib/datadog_flutter_plugin_web.dart

```diff
@@ -16,6 +17,19 @@ import 'src/logs/ddlogs_web.dart';
 import 'src/rum/ddrum_platform_interface.dart';
 import 'src/rum/ddrum_web.dart';
 
+@anonymous
+extension type JsUser._(JSObject _) implements JSObject {
```

Comment:
nit: I would reference the original interface, as you did in https://github.com/DataDog/dd-sdk-flutter/pull/787

```suggestion
/// Mirrors the `User` interface from the Datadog Browser SDK: 
/// https://github.com/DataDog/browser-sdk/blob/562214cd1457e5d8445c1cb52424d9615f892158/packages/core/src/domain/contexts/userContext.ts#L10
///
/// Keep this class updated to match the original interface.
@anonymous
extension type JsUser._(JSObject _) implements JSObject {
```

Reply:
I may have to go back and do this with all structures, as they all have Browser SDK equivalents. I'll have to think about it...